### PR TITLE
Fixes related to #181 and #180

### PR DESCRIPTION
**Pull request by edgarmueller**

### CODE REVIEW

File: components/form/form-directive.ts [hunk 1]
```diff
@@ -90,15 +90,11 @@ class FormController {
 
     private fetchSchema() {
         if (typeof this.scope.schema === "object") {
-            var p: ng.IDeferred<any> = this.$q.defer<any>();
-            p.resolve(this.scope.schema);
-            return p;
+            return this.$q.when(this.scope.schema);
```
Comment (sdirix) [inline components/form/form-directive.ts]: Nice! Way more readable than before

File: components/renderers/controls/array/array-renderer.ts [hunk 2]
```diff
@@ -120,10 +120,14 @@ class ArrayRenderer implements JSONForms.IRenderer {
...
                 containerDescription['submitControls'] = this.createControlsForSubmit(items, schemaPath, submitElement, services);
+                if (data == undefined) {
+                    containerDescription["instance"][containerDescription['path']] = [];
+                    data = containerDescription["instance"][containerDescription['path']];
+                }
                 containerDescription['submitCallback'] = () => data.push(_.clone(submitElement));
             }
             containerDescription['generateControlDescriptions'] = (data) =>
-                this.generateControlDescriptions(items, schemaPath, containerDescription['elements'], data.length, services);
+                this.generateControlDescriptions(items, schemaPath, containerDescription['elements'],  data.length, services);
```
Comment (sdirix) [inline components/renderers/controls/array/array-renderer.ts]: We probably should add some automatic formatter

File: components/renderers/controls/array/array-renderer.ts [hunk 3]
```diff
@@ -120,10 +120,14 @@ class ArrayRenderer implements JSONForms.IRenderer {
             if (supportsSubmit) {
                 containerDescription['submitElement'] = submitElement;
                 containerDescription['submitControls'] = this.createControlsForSubmit(items, schemaPath, submitElement, services);
+                if (data == undefined) {
+                    containerDescription["instance"][containerDescription['path']] = [];
+                    data = containerDescription["instance"][containerDescription['path']];
```
Comment (sdirix) [inline components/renderers/controls/array/array-renderer.ts]: Replacing the second line with ```data = []``` is easier to read and understand

**Comment by sdirix** [comment]
Nice! I especially like the changes within the form directive.

If you have time it would be great if you fix the two minor nitpicks within the array renderer ;) Thx!

File: components/renderers/controls/array/array-renderer.ts (hunk 2 above)
Comment (edgarmueller) [inline components/renderers/controls/array/array-renderer.ts]: I fully agree, this is already tracked via https://github.com/eclipsesource/jsonforms/issues/105

**Comment by sdirix** [comment]
Great work!

**Pull request merged**